src/ucp/core/ucp_request.c

```diff
@@ -298,6 +314,14 @@ int ucp_request_pending_add(ucp_request_t *req)
                 ucs_debug_get_symbol_name(req->send.uct.func));
 
     uct_ep = req->send.ep->uct_eps[req->send.lane];
+    if ((req->flags & UCP_REQUEST_FLAG_SEND_AM) &&
```

Comment:
Both is "req->flags" and "req->send.msg_proto.am.flags"?
If yes it's because I thought that ucp_request_pending_add can possibly be called today or in the future from other flows that are not AM,
And if it happens checking only "req->send.msg_proto.am.flags" can cause nasty bugs... (union)

Reply:
I mean, whether all 4 combinations of {`UCP_REQUEST_FLAG_SEND_AM`, `UCP_AM_SEND_FLAG_COPY_HEADER`} x {0,1} are possible?
If yes - how?
If no - why check 2 conditions then?